feat: CoAP response/request clarification

jjmilburn opened the pull request:
`content-format` is only relevant for responses, and URI queries (if
present) only apply to requests.

Also, add a few simple `GET` examples to show the on-the-wire header
format representation.

Commit:
feat: CoAP response/request clarification

`content-format` is only relevant for responses, and URI queries (if
present) only apply to requests.

Also, add a few simple `GET` examples to show the on-the-wire header
format representation.

ricehornet commented on coap_format_spec.md:
```diff
@@ -92,13 +92,16 @@ urther defining the resource. In this way, the URI may *roughly* mirror the
 not required.
 
-The following option type *should* be present when sending content:
+(There may be multiple `uri-path` options present, e.g. 'batt/main' and
```
This is a bit misleading when reading the examples in the Appendix. It seems like when there are multiple levels to the path, that there is one `uri-path` option per level when serialized (for example, one for `batt` and one for `main`). This is what I think needs to be clear here, as well as that all the examples given already are valid `uri-path`s.

jjmilburn replied:
ok, I will try to clarify here.

ricehornet commented on coap_format_spec.md:
```diff
@@ -189,3 +192,29 @@ which can speed up resource discovery, but is not strictly required.
+* `b4` - Option delta '0xb' (11, `uri-path`), length 4
+* `62 61 74 74` - the value for `uri-path`, which is 'batt' in ASCII
+* `04` - Option delta '0x0` (0, still `uri-path`), length 4
+* `6d 61 69 6e` - the vlaue for `uri-path, which is `main` in ASCII
```
```suggestion
* `6d 61 69 6e` - the value for `uri-path, which is `main` in ASCII
```

ricehornet reviewed (approved):
@jjmilburn I think that the part about `uri-path` could be improved -- as I was a bit confused by it. Otherwise looks good to me.

Commit:
fix: Clarify `uri-path` option, fix typo

`uri-path` is present at least one time, but may be present multiple
times depending on the length of the `uri-path`.

jjmilburn commented:
@ricehornet OK, I think comments are addressed!

ricehornet commented:
Great, looks good. +1!
